components/connector/internal/revocation/loader.go

```diff
@@ -0,0 +1,109 @@
+package revocation
+
+import (
+	"context"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+type Loader interface {
+	Run(ctx context.Context)
+}
+
+type revocationListLoader struct {
+	revocationListCache Cache
+	manager             Manager
+	configMapName       string
+	reconnectInternal   time.Duration
+}
+
+func NewRevocationListLoader(revocationListCache Cache,
+	manager Manager,
+	configMapName string,
+	reconnectInternal time.Duration,
+) Loader {
+	return &revocationListLoader{
+		revocationListCache: revocationListCache,
+		manager:             manager,
+		configMapName:       configMapName,
+		reconnectInternal:   reconnectInternal,
+	}
+}
+
+func (rl *revocationListLoader) Run(ctx context.Context) {
+	rl.startKubeWatch(ctx)
+}
+
+func (rl *revocationListLoader) startKubeWatch(ctx context.Context) {
```

Comment:
Is this function needed you can add the logic in Run since Run only invokes this function and does not give any benefit.

Reply:
cleaner name as it will show exactly what will be started, not sure if it makes any sense

Reply:
It sounds reasonable I am ok with it.